### ISSUE TITLE
Constant Memory leak still persists after upgrading 1.5.1.834 to 1.5.4.858

### ISSUE THREAD

**Issue by fxcebx** [rstudio/shiny-server#315]
I created a flexdashboard with eight renderPlot(.. ggplot...) and three renderUI(... htmlTable...).
Data is updated every 30 seconds with reactiveFileReader using feather format. 

Shiny-server runs on a virtual Ubuntu 16.04.02 LTS with 4 cores and 8 GByte.
For better CPU load spread I run the same app four times, which gives me four R processes. In total 12-18 Chrome sessions (9 machines, rest is office users) are about equally connected with the four apps. It really works great, but constantly keeps eating up memory by 1.5 GByte every two days (same for 1.5.1 and 1.5.4).

![image](https://user-images.githubusercontent.com/5912082/29462027-325d4116-842e-11e7-8f01-e3560503c324.png)

The smaller memory usage drops occur, if I kill a single R PID or two. The big drops are seen, either by restarting shiny-server or killing all four R processes. It frees the same amount of memory if I kill all R processes or a shiny-server restart.

If it were not that the dashboards are shown at production machines, I would not bother regularly killing the memory increased R processes. Its for someone to press the Reload buttons, after the Chrome browsers loose their connections.

Maybe I misplaced this issue wrongly with shiny-server and it's related with shiny. 
Any ideas how to further debug?

**Comment by jcheng5** [comment]
Just to be clear, this graph is the memory usage for the whole machine? Or just for the shiny-server process itself? i.e. is the memory leak in the R processes (which would make sense to me) or in shiny-server (which would be more surprising)?

**Comment by fxcebx** [comment]
You are right, its the memory usage of R processes shown, as with Monitorix I check here for R command. I included actual sudo top for comparison.  

![image](https://user-images.githubusercontent.com/5912082/29484444-7ee839b4-84f1-11e7-9568-88fb7271d38c.png)


![image](https://user-images.githubusercontent.com/5912082/29484367-7de5332a-84ef-11e7-9f59-09144f337f25.png)
